examples/23_flag_subcommands_pacman.rs

```diff
@@ -0,0 +1,153 @@
+// Working with flag subcommands allows behavior similar to the popular Archlinux package manager Pacman.
+// Man page: https://jlk.fjfi.cvut.cz/arch/manpages/man/pacman.8
+//
+// It's suggested that you read examples/20_subcommands.rs prior to learning about `FlagSubCommand`s
+//
+// This differs from normal subcommands because it allows passing subcommand as `clap::Arg` in short or long args.
+//
+//            Top Level App (pacman)                              TOP
+//                           |
+//    ---------------------------------------------------
+//   /     |        |        |         |        \        \
+// sync  database remove    files      query   deptest   upgrade  LEVEL 1
+//
+// Given the above hierachy, valid runtime uses would be (not an all inclusive list):
+//
+// $ pacman -Ss
+//           ^--- subcommand followed by an arg in its scope.
+//
+// $ pacman -Qs
+//
+// $ pacman -Rns
+//
+// NOTE: Subcommands short flags don't have to be uppercase.
```

Comment:
You mean they can be either uppercase or lowercase?

Reply:
Yeah I did, thanks!